lib/liquid/combinators/tag.ex

```diff
@@ -5,33 +5,53 @@ defmodule Liquid.Combinators.Tag do
   import NimbleParsec
 
   @doc """
-  Define a tag from a tag_name and a function. The function `combinator` must expect
-  a combinator and must returns a combinator
+  Define a tag from a tag_name and, optionally, a function to parse tag parameters,
+  an end_tag_name to close the tag and a function to parse the body inside the tag
+  Both functions must receive a combinator and must return a combinator
 
   The returned tag is a combinator which expect a start tag `{%` a tag name and a end tag `%}`
 
-  The second parameter, function combinator, contains the specific behavior for defined tag
-
   ## Examples
 
       defmodule MyParser do
         import NimbleParsec
         alias Liquid.Combinators.Tag
 
         def ignorable, do: Tag.define(
-          :ignorable,
+          "ignorable",
           fn combinator -> combinator |> string("T") |> ignore() |> integer(2,2))
 
       MyParser.ignorable("{% ignorable T12 %}")
       #=> {:ok, {:ignorable, [12]}, "", %{}, {1, 0}, 2}
   """
-  def define(tag_name, combinator \\ & &1) do
+  def define(tag_name, combinator_open \\ & &1, end_tag_name \\ "", combinator_body \\ & &1) do
```

Comment:
Isn't the end_tag_name always `end+tag_name`?

Reply:
There are 1 exception, else is not elseif, it is elsif.